Add docs for --device option in Windows (#1663)
docker/cli: docs/reference/commandline/run.md

> that may be removed should not be added to untrusted containers with
> `--device`.

> For Windows, the format of the string passed to the `--device` option is in the form of `--device=<IdType>/<Id>`. Today, Windows only supports an IdType of `class` and the Id as a [device interface class GUID](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/overview-of-device-interface-classes). For a list of container-supported device interface class GUIDs, please refer to the table defined in the [Windows container docs](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/hardware-devices-in-containers).

Reviewer: Rather than `Today`, perhaps state what versions of Windows. (AFAIK, this is RS5/1809 on only).

Author: Fixed

> For Windows, the format of the string passed to the `--device` option is in the form of `--device=<IdType>/<Id>`. Beginning with Windows Server 2019 and Windows 10 October 2018 Update, Windows only supports an IdType of `class` and the Id as a [device interface class GUID](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/overview-of-device-interface-classes). For a list of container-supported device interface class GUIDs, please refer to the table defined in the [Windows container docs](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/hardware-devices-in-containers).

Reviewer 2: Trying to avoid "please" in the documentation; perhaps 

```Markdown
Refer to the table defined in the [Windows container
documentation](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/hardware-devices-in-containers)
for a list of container-supported device interface class GUIDs.
```

Also, could you wrap these to ~80 chars?

```suggestion
For Windows, the format of the string passed to the `--device` option is in
the form of `--device=<IdType>/<Id>`. Beginning with Windows Server 2019
and Windows 10 October 2018 Update, Windows only supports an IdType of
`class` and the Id as a [device interface class
GUID](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/overview-of-device-interface-classes).
Refer to the table defined in the [Windows container
documentation](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/hardware-devices-in-containers)
for a list of container-supported device interface class GUIDs.
```

Author: Thanks, added these fixes.

> When this option is specified for a process-isolated Windows container, _all_ devices which implement the requested device interface class GUID will be made available in the container. For example, the command below will make all COM ports on the host visibile in the container.

Reviewer 2: Should it be "devices _that_ implement" (instead of "which"?)

Also, try to avoid future tense in technical documentation; here's a quick go at changing it to remove the future tense, and wrap the paragraph to 80 chars;

```suggestion
If this option is specified for a process-isolated Windows container, _all_
devices that implement the requested device interface class GUID are made
available in the container. For example, the command below makes all COM
ports on the host visible in the container.
```

Author: You are correct--I've now learned when to use 'which' vs 'that'. It always confounded me in the past.

```powershell
PS C:\> docker run --device=class/86E0D1E0-8089-11D0-9CE4-08003E301F73 mcr.microsoft.com/windows/servercore:ltsc2019 
```

> > **Note**: `--device` may only be used on Windows with process-isolated Windows containers. This option will fail if the container

Reviewer 2: Slightly changed to wrap to 80-chars, and remove future tense;

```suggestion
> **Note**: the `--device` option is only supported on process-isolated
> Windows containers. This option fails if the container isolation is `hyperv`
> or when running Linux Containers on Windows (LCOW).
```

Author: Added

> that may be removed should not be added to untrusted containers with
> `--device`.

For Windows, the format of the string passed to the `--device` option is in
the form of `--device=<IdType>/<Id>`. Beginning with Windows Server 2019
and Windows 10 October 2018 Update, Windows only supports an IdType of
`class` and the Id as a [device interface class

> GUID](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/overview-of-device-

Reviewer 2: oh; the URL itself cannot be wrapped (it's ok for the caption, but not for the URL)

interface-classes).
Refer to the table defined in the [Windows container

> docs](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy

Reviewer 2: same here
